src/components/BookUpdate.tsx

```diff
@@ -0,0 +1,103 @@
+"use client";
+import AutoForm from "@/components/ui/auto-form";
+import * as z from "zod";
+import { Card, CardContent, CardHeader, CardTitle } from "./ui/card";
+import { Button } from "@/components/ui/button";
+import { api } from "~/utils/api";
+import React from "react";
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/components/ui/popover";
+
+const BookInputSchema = z.object({
+  title: z
+    .string({
+      required_error: "Title is required.",
+    })
+    .min(2, {
+      message: "Title must be at least 2 characters.",
+    }),
+
+  author: z
+    .string({
+      required_error: "Author is required.",
+    })
+    .min(2, {
+      message: "Author must be at least 2 characters.",
+    }),
+});
+
+const BookUpdate = ({ bookId }: { bookId: string }): JSX.Element => {
+  const { data: queryData, refetch: refetchBook } =
```

Comment:
Rename

Reply:
Is this because there are other functions basically named the same thing?

Reply:
Query data is just unclear